### PR TITLE
Inject line number into the rendered markdown HTML

### PR DESCRIPTION
**Pull request by amoshydra**
Done a similar implementation like the markdown-it demo page
https://github.com/markdown-it/markdown-it/blob/master/support/demo_template/index.js#L165

For now it will only track the line number for paragraph and header elements at the level 1 hierarchy.
These should be sufficient for implementing a very basic scroll sync mechanism.

Other level 1 hierarchy element that are not included in this implementation includes
- [ ] hr
- [ ] blockquote
- [ ] pre (code block, diagrams, ascii-math)

### CODE REVIEW

**Comment by amoshydra** [comment]
@chuajiaxuan will need your help in updating the test data.

Update: I have updated the test data in a separate PR #390. If the changed is ok, you may merge the #390 into this PR.

**Comment by chuajiaxuan** [comment]
@amoshydra  Approved PR #390  already!

File: sashimi-webapp/src/logic/documentPackager/markdownProcessor.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ import mdAsciiMath from './markdown-it-asciimath';
 import mdDiagrams from './markdown-it-diagram';
 // Custom conditional plugin
 import mdConditional from './conditionalProcessor';
+import mdLineNumber from './plugins/markdown-it-line-number';
```
Comment (chuajiaxuan) [inline sashimi-webapp/src/logic/documentPackager/markdownProcessor.js:24]: @amoshydra  Should we refactor and import all self-written/customised plugins from plugins folder?

**Review by chuajiaxuan — APPROVED** [review]
Nice job with the synchronised scrolling! Looks good to go!

File: sashimi-webapp/src/logic/documentPackager/markdownProcessor.js (hunk 1 above)
Comment (amoshydra) [reply sashimi-webapp/src/logic/documentPackager/markdownProcessor.js:24]: yeah, that should be the preferred way, grouping all the self-written plugins into the plugins folder
- conditionalProcessor.js
- markdown-it-asciimath.js
- markdown-it-diagram.js

**Comment by amoshydra** [comment]
@chuajiaxuan This pr does not contain code for scroll sync yet. Instead, it is a requirement for an upcoming implementation of scroll sync.

**Pull request merged**